src/foam/nanos/boot/NSpec.js

```diff
@@ -217,28 +270,69 @@ foam.CLASS({
           return service;
         }
 
-        Interpreter shell = new Interpreter();
-        try {
-          shell.set("x", x);
-          Object service = shell.eval(getServiceScript());
-          saveService(x, service);
-          return service;
-        } catch (EvalError e) {
-          foam.nanos.logger.Logger logger = (foam.nanos.logger.Logger) x.get("logger");
-          if ( logger != null ) {
-            logger.error("NSpec serviceScript error", getServiceScript(), e);
-          } else {
-            System.err.println("NSpec serviceScript error: " + getServiceScript());
-            e.printStackTrace();
+        Language l = getLanguage();
+        if ( l == foam.nanos.script.Language.JSHELL ) {
+          JShell jShell = JShell
+            .builder()
+            .out(ps)
+            .executionEngine(new ExecutionControlProvider() {
+              @Override
+              public String name() {
+                return "direct";
+              }
+
+              @Override
+              public ExecutionControl generate(ExecutionEnv ee, Map<String, String> map) throws Throwable {
+                return new DirectExecutionControl();
+              }
+            }, null)
+            .build();
+
+          foam.nanos.script.Script.X_HOLDER[0] = x;
+          String init = "import foam.core.X; import foam.nanos.boot.NSpec; X x = foam.nanos.script.Script.X_HOLDER[0]; ";
+
+          BufferedReader rdr = new BufferedReader(new StringReader(init+getServiceScript()));
+          List<String> l1 = new ArrayList<String>();
+
+          for ( String line = rdr.readLine(); line != null; line = rdr.readLine() ) {
+            l1.add(line);
+          }
+
+          List<String> instructionList = new InstructionPresentation(jShell).parseToInstruction(l1);
+          EvalInstruction console = new EvalInstruction(jShell, instructionList, x);
+          String print = null;
+          print = console.runEvalInstruction();
```

Comment:
eliminate the extra line .. 
```suggestion
          String print = console.runEvalInstruction();
```